spec/tunes/app_version_spec.rb

```diff
@@ -150,10 +171,308 @@
+      let(:ipad_trailer_path) { "path_to_trailer.mov" }
+      let(:ipad_trailer_preview_path) { "path_to_trailer_preview.jpg" }
+      let(:ipad_external_valid_trailer_preview_path) { "path_to_my_screenshot.jpg" }
+      let(:ipad_external_invalid_trailer_preview_path) { "path_to_my_invalid_screenshot.jpg.jpg" }
+      before do
+        allow(Spaceship::UploadFile).to receive(:from_path) do |path|
+          r = du_uploadtrailer_correct_mov if path == ipad_trailer_path
+          r = du_uploadtrailer_preview_correct_jpg if path == ipad_trailer_preview_path
+          r = du_uploadtrailer_preview_correct_jpg if path == ipad_external_valid_trailer_preview_path
+          r
+        end
+
+        allow(Spaceship::Utilities).to receive(:grab_video_preview) do |path|
+          r = ipad_trailer_preview_path if path == ipad_trailer_path
+          r
+        end
+
+        allow(Spaceship::Utilities).to receive(:portrait?) do |path|
+          r = true if path == ipad_trailer_preview_path
+          r = true if path == ipad_external_invalid_trailer_preview_path
+          r = true if path == ipad_external_valid_trailer_preview_path
+          r
+        end
+
+        allow(Spaceship::Utilities).to receive(:resolution) do |path|
+          r = [900, 1200] if path == ipad_trailer_path
+          r = [768, 1024] if path == ipad_trailer_preview_path
+          r = [700, 1000] if path == ipad_external_invalid_trailer_preview_path
+          r = [768, 1024] if path == ipad_external_valid_trailer_preview_path
+          r
+        end
+
+        json = JSON.parse(du_read_fixture_file("upload_trailer_response_success.json"))
+        allow(client.du_client).to receive(:upload_trailer).and_return(json)
+
+        json = JSON.parse(du_read_upload_trailer_preview_response_success)
+        allow(client.du_client).to receive(:upload_trailer_preview).and_return(json)
+      end
+
+      def trailers(device)
+        version.trailers["English"].find_all { |s| s.device_type == device }
+      end
+
+      def ipad_trailers
+        trailers("ipad")
+      end
+
+      it "cannot add a trailer to iphone35" do
+        expect do
+          version.upload_trailer!(ipad_trailer_path, "English", 'iphone35')
+        end.to raise_error "No app trailer supported for iphone35"
+      end
+
+      it "requires timestamp with a specific format" do
+        expect do
+          version.upload_trailer!(ipad_trailer_path, "English", 'ipad', "00:01.000")
+        end.to raise_error "Invalid timestamp 00:01.000"
+        expect do
+          version.upload_trailer!(ipad_trailer_path, "English", 'ipad', "01.000")
+        end.to raise_error "Invalid timestamp 01.000"
+      end
+
+      it "can add a new trailer" do
+        # remove existing
+        version.upload_trailer!(nil, "English", 'ipad')
+
+        count = ipad_trailers.count
+        expect(count).to eq(0)
+        version.upload_trailer!(ipad_trailer_path, "English", 'ipad')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.